src/gui/widgets/chatbox.cpp

```diff
@@ -587,23 +596,27 @@ void chatbox::process_message(const ::config& data, bool whisper /*= false*/)
 	if(whisper) {
 		add_whisper_received(sender, message);
 	} else {
-		if (!prefs::get().parse_should_show_lobby_join(sender, message)) return;
 
-		std::string room = data["room"];
+		if (!prefs::get().parse_should_show_lobby_join(sender, message)) return;
 
-		// Attempt to send to the currently active room first.
-		if(room.empty()) {
-			LOG_LB << "Message without a room from " << sender << ", falling back to active window";
-			room = open_windows_[active_window_].name;
-		}
+		std::string room;
+		std::string type = data["type"];
 
-		// If we still don't have a name, fall back to lobby.
-		if(room.empty()) {
-			LOG_LB << "Message without a room from " << sender << ", assuming lobby";
+		if((sender == "server" && type == "motd") || sender != "server") {
```

Comment:
Server messages do not go into the lobby unless it's the motd? What is this logic based on?

Reply:
This is the implementation for the idea described in my comment. It tries to separate responses to commands that the user runs so that they are displayed in the current room and are visible.